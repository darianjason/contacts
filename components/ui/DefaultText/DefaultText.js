// @flow

import * as React from 'react';
import {Text} from 'react-native';
import type {TextStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';

import styles from './DefaultText.styles';

type Props = {
  style?: TextStyleProp,
  numberOfLines?: number,
  children: ?React.Node,
};

const DefaultText = ({style, numberOfLines, children}: Props): React.Node => (
  <Text style={[styles.text, style]} numberOfLines={numberOfLines}>
    {children}
  </Text>
);

export default DefaultText;
