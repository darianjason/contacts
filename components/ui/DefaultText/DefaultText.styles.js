import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants';

const styles = StyleSheet.create({
  text: {
    color: Colors.black,
    fontFamily: 'Manrope-Regular',
  },
});

export default styles;
