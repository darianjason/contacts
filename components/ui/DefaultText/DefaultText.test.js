import React from 'react';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import DefaultText from './DefaultText';

describe('Default Text component', () => {
  test('renders text', () => {
    const {getByText} = render(<DefaultText>Test</DefaultText>);

    expect(getByText('Test')).toBeTruthy();
  });

  test('renders correctly', () => {
    const tree = renderer.create(<DefaultText>Test</DefaultText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
