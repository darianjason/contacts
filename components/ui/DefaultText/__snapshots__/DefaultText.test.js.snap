// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Text component renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#060A0E",
      "fontFamily": "Manrope-Regular",
    }
  }
>
  Test
</Text>
`;
